pkg/datapath/linux/neigh/neigh.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2019 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package neigh
+
+import (
+	"net"
+
+	"github.com/cilium/cilium/pkg/datapath/linux/route"
+
+	"github.com/cilium/arping"
+	"github.com/sirupsen/logrus"
+	"github.com/vishvananda/netlink"
+)
+
+func NeighAddAddress(address net.IP) error {
+	link, err := route.NodeDeviceWithDefaultRoute()
+	if err != nil {
+		logrus.WithError(err).Error("Unable to retrieve default route device")
+		return err
+	}
+	// XXX arping lib sucks in that we need such lookup twice
+	iface, err := net.InterfaceByName(link.Attrs().Name)
```

Comment:
Shouldn't we do a lookup via `option.Config.Device` in the case we pass `--device=<..>` to cilium-agent?

Reply:
Agree, true. I'll fix.